src/libcrun/utils.c

```diff
@@ -897,8 +897,23 @@ libcrun_is_apparmor_enabled (libcrun_error_t *err)
   return apparmor_enabled;
 }
 
+static int 
+is_current_process_confined(libcrun_error_t *err)
+{
+  int fd = open("/proc/thread-self/attr/current", O_RDONLY|O_CLOEXEC);
+  
+  if (fd != -1) {
+    char buf[256];
+    ssize_t bytes_read = read(fd, buf, sizeof(buf) - 1);
+    close(fd);
+    if (bytes_read != -1)
+      return (strncmp(buf, "unconfined", bytes_read) != 0 && buf != '\0');
```

Comment:
Possible return values from this return statement are 0 and 1.
```
return (strncmp(buf, "unconfined", bytes_read) != 0 && buf != '\0');
```
but the caller checks if the result is negative.

```
      ret = is_current_process_confined(err);
      if (UNLIKELY (ret < 0))
        return ret;
```

Reply:
True, but that is the good case with ret value 0 and 1.
In case of error negative value is returned.

Reply:
Oops, I missed that `ret` is actually checked here too:
```
 xasprintf (&buf, "%s %s", no_new_privileges && ret ? "stack": now ? "changeprofile" : "exec", profile);
```
Sorry

Reply:
Another thing:
Regarding
```
   return (strncmp(buf, "unconfined", bytes_read) != 0 && buf != '\0');
```

Did you mean
```
   return (strncmp(buf, "unconfined", bytes_read) != 0 && buf[0] != '\0');
```
?
(It looks like you wanted to check if `buf` contains an empty string)

Reply:
oh yes, thanks.